core/templates/pages/collection-editor-page/navbar/collection-editor-navbar-breadcrumb.component.spec.ts

```diff
@@ -0,0 +1,141 @@
+import { CollectionEditorRoutingService } from '../services/collection-editor-routing.service';
+import { FocusManagerService } from 'services/stateful/focus-manager.service';
+import { Collection } from 'domain/collection/collection.model';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+
+describe('Collection editor navbar breadcrumb component', () => {
+  let collectionEditorStateService: CollectionEditorStateService;
+  let collectionEditorRoutingService: CollectionEditorRoutingService;
+  let focusManagerService: FocusManagerService;
+  let component: CollectionEditorNavbarBreadcrumbComponent;
+  let fixture: ComponentFixture<CollectionEditorNavbarBreadcrumbComponent>;
+  let mockCollection: Collection;
+
+  let activeTab = 'edit';
+  let collectionData = {
+    id: 'collection_id',
+    title: 'collection title',
+    objective: 'collection objective',
+    languageCode: 'en',
+    tags: ['mock tag'],
+    playthrough: null,
+    category: 'Collection Category',
+    version: 0,
+    schemaVersion: 1,
+    nodes: []
+  };
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule
+      ],
+      declarations: [
+        CollectionEditorNavbarBreadcrumbComponent
+      ],
+      providers: [
+        CollectionEditorStateService,
+        CollectionEditorRoutingService,
+        FocusManagerService
+      ],
+      schemas: [
+        NO_ERRORS_SCHEMA
+      ]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(
+      CollectionEditorNavbarBreadcrumbComponent);
+    component = fixture.componentInstance;
+    collectionEditorStateService = TestBed.inject(CollectionEditorStateService);
+    collectionEditorStateService = (collectionEditorStateService as unknown) as
+      jasmine.SpyObj<CollectionEditorStateService>;
+    collectionEditorRoutingService = TestBed.inject(
+      CollectionEditorRoutingService);
+    collectionEditorRoutingService = (
+      (collectionEditorRoutingService as unknown) as
+        jasmine.SpyObj<CollectionEditorRoutingService>);
+    focusManagerService = TestBed.inject(FocusManagerService);
+    focusManagerService = (focusManagerService as unknown) as
+      jasmine.SpyObj<FocusManagerService>;
+  });
+
+  beforeEach(() => {
+    mockCollection = new Collection(
+      collectionData.id, collectionData.title, collectionData.objective,
+      collectionData.languageCode, collectionData.tags,
+      collectionData.playthrough, collectionData.category,
+      collectionData.version, collectionData.schemaVersion,
+      collectionData.nodes);
+  });
+
+  it('should load the component properly', () => {
+    spyOn(collectionEditorStateService, 'getCollection').and.returnValue(
+      mockCollection);
+    spyOn(collectionEditorRoutingService, 'getActiveTabName').and.returnValue(
+      activeTab);
+
+    component.ngOnInit();
+    fixture.whenStable()
+      .then(() => {
```

Comment:
Do we need this? Ditto Everywhere

Reply:
I have removed it from most of the places but it is required in `collection-footer` for the 2nd it statement I guess. Because removing it causes that test to fail